src/execution/compiler/operator/seq_scan_translator.cpp

```diff
@@ -215,6 +217,8 @@ void SeqScanTranslator::ScanVPI(WorkContext *ctx, FunctionBuilder *function, ast
       function->Append(assign);
       // Push to parent.
       ctx->Push(function);
+
+      CounterAdd(function, num_scans_, 1);
```

Comment:
This is a small concern. `num_scans_` records the number of tuples that match the filter and are passed on to the next translator. This means that as far as the model is concerned, the cost of scanning 1 million tuples to emit 1 tuple is the same as the cost of scanning 1 tuple to emit 1 tuple.

Is there a way to get the raw number of tuples in the table (or at least a close approximation)?

Reply:
I think the concern is valid. Maybe directly utilize the `SqlTable::GetNumTuple` function?

Reply:
Changed it to add the tuple count from the VPI.

For the record, looks like `SqlTable::GetNumTuple` is borked for getting approximate counts.

Reply:
I think your change works.